app/services/metrc_service/plant/discard.rb

```diff
@@ -2,56 +2,67 @@ module MetrcService
   module Plant
     class Discard < MetrcService::Base
       NOT_SPECIFIED = 'Not Specified'.freeze
+      METHOD_CALL = {
+        immature: :destroy_immature_plants,
+        mature: :destroy_mature_plants
+      }.freeze
 
       def call
-        payload = send("build_#{plant_state}_payload", discard, batch)
-        action = plant_state == 'immature' ? :destroy_plant_batches : :destroy_plants
-
-        call_metrc(action, payload)
-
+        send(METHOD_CALL[plant_state])
```

Comment:
why this double indirection of a send + hash +two methods, instead of a simple if/else with direct calls to `call_metrc`?

It requires more lines, and obscures the logic.